app/cdap/components/AddTransformation/SelectColumnsList/index.tsx

```diff
@@ -0,0 +1,247 @@
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Radio,
+  Typography,
+} from '@material-ui/core';
+import React, { useEffect, useRef, useState } from 'react';
+import { useStyles } from '../styles';
+import { SearchIcon } from '../iconStore';
+import { prepareDataQualtiy } from '../CircularProgressBar/utils';
+import DataQualityProgress from '../CircularProgressBar';
+import { NoDataSVG } from 'components/GridTable/iconStore';
+import T from 'i18n-react';
+
+export default function(props) {
+  const {
+    directiveFunctionSupportedDataType,
+    selectedColumnsCount,
+    columnData,
+    setSelectedColumns,
+    dataQuality,
+  } = props;
+  const [columns, setColumns] = useState(columnData);
+  const [dataQualityValue, setDataQualityValue] = useState(dataQuality);
+  const [selectedColumns, setSelectedColumn] = useState([]);
+  const [focused, setFocused] = useState(false);
+  const classes = useStyles();
+  const ref = useRef(null);
+
+  const no_match =
+    directiveFunctionSupportedDataType?.length > 0 &&
+    directiveFunctionSupportedDataType?.includes('all')
+      ? directiveFunctionSupportedDataType?.filter((el) => el == 'all')
+      : columns.filter((object1) => {
+          return directiveFunctionSupportedDataType?.some((object2) => {
+            return object2.includes(object1?.type[0]?.toLowerCase());
+          });
+        });
+
+  useEffect(() => {
+    const getPreparedDataQuality = prepareDataQualtiy(dataQuality, columnData);
+    setDataQualityValue(getPreparedDataQuality);
+  }, []);
+
+  const onSelect = (column) => {
+    setSelectedColumns([column]);
+    setSelectedColumn([column]);
+  };
+
+  const handleSearch = (event) => {
+    if (event.target.value) {
+      const columnValue = columnData.filter((el) =>
+        el?.label.toLowerCase().includes(event.target.value.toLowerCase())
+      );
+      if (columnValue.length) {
+        setColumns(columnValue);
+      } else {
+        setColumns([]);
+      }
+    } else {
+      setColumns(columnData);
+    }
+  };
+
+  const handleFocus = () => {
+    ref?.current.focus();
+    setFocused(true);
+  };
+
+  return (
+    <section className={classes.columnsCountTextStyles}>
+      <div className={classes.selectColumnsHeaderStyles}>
+        <div>
+          {selectedColumnsCount
+            ? selectedColumnsCount > 10
+              ? selectedColumnsCount
+              : `0${selectedColumnsCount}`
+            : 'No '}{' '}
+          &nbsp;{T.translate('features.WranglerNewAddTransformation.columnsSelected')}
+        </div>
+        <div className={classes.searchFormControl}>
+          <input
+            className={focused ? classes.isFocused : classes.isBlurred}
+            onChange={handleSearch}
+            ref={ref}
+            onFocus={() => setFocused(true)}
+            onBlur={() => setFocused(false)}
+          />
+          <Box className={classes.searchInputAdornment} onClick={handleFocus}>
+            {SearchIcon()}
```

Comment:
this icon can be used as a component here, plz update the code, it cannot be called as a function in dom code

Reply:
use it like NoDataSVG

Reply:
Done